portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/site/LessonsTreeView.java

```diff
@@ -0,0 +1,262 @@
+        return ("SELECT p.toolId as sakaiPageId," +
+            " p.pageId as lessonsPageId," +
+            " s.site_id as sakaiSiteId," +
+            " s.tool_id as sakaiToolId," +
+            " i.id as lessonsItemId," +
+            " i.name as name," +
+            " i.description as description," +
+            " " + toChar(conn, "i.sakaiId") + " as sendingPage," +
+            " p2.hidden as hidden," +
+            " p2.releaseDate as releaseDate," +
+            " log.complete as completed," +
+            " i.required," +
+            " i.prerequisite" +
+            " FROM lesson_builder_pages p" +
+            " INNER JOIN sakai_site_tool s" +
+            "   on p.toolId = s.page_id" +
+            " INNER JOIN lesson_builder_items i" +
+            "   on (i.pageId = p.pageId AND type = 2)" +
+            " INNER JOIN lesson_builder_pages p2" +
+            "   on (p2.pageId = i.sakaiId)" +
+            " LEFT OUTER JOIN lesson_builder_log log" +
+            "   on (log.itemId = i.id AND log.userId = ?)" +
+            " WHERE p.parent IS NULL" +
+            "   AND p.toolId in (" + placeholdersFor(pageIds) + ")" +
+            " ORDER BY i.sequence");
+    }
+
+    private String toChar(final Connection conn, final String expr) {
+        try {
+            final String dbFamily = conn.getMetaData().getDatabaseProductName().toLowerCase(java.util.Locale.ROOT);
+
+            if ("mysql".equals(dbFamily)) {
+                return String.format("cast(%s AS CHAR)", expr);
+            } else if ("oracle".equals(dbFamily)) {
+                return String.format("to_char(%s)", expr);
+            } else {
+                return expr;
+            }
+        } catch (SQLException e) {
+            return expr;
+        }
+    }
+
+    private Map<String, String> makeMap(final ResultSet rs) throws SQLException {
+        final Map<String, String> result = new HashMap<>();
+
+        result.put("toolId", rs.getString("sakaiToolId"));
+        result.put("siteId", rs.getString("sakaiSiteId"));
+        result.put("itemId", rs.getString("lessonsItemId"));
+        result.put("sendingPage", rs.getString("sendingPage"));
+        result.put("name", rs.getString("name"));
+        result.put("description", rs.getString("description"));
+        result.put("hidden", rs.getInt("hidden") == 1 ? "true" : "false");
+
+        result.put("required", rs.getInt("required") == 1 ? "true" : "false");
+        result.put("completed", rs.getInt("completed") == 1 ? "true" : "false");
+        result.put("prerequisite", rs.getInt("prerequisite") == 1 ? "true" : "false");
+
+        if (rs.getTimestamp("releaseDate") != null) {
+            final Timestamp releaseDate = rs.getTimestamp("releaseDate");
+            if (releaseDate.getTime() > System.currentTimeMillis()) {
+                result.put("hidden", "true");
+                final DateFormat df = new SimpleDateFormat("MMM d, yyyy h:mm a");
```

Comment:
If this is just for user display, more i18n-friendly way would be via a predefined format that uses the preferred locale: https://docs.oracle.com/javase/tutorial/i18n/format/dateFormat.html